dataprofiler/profilers/profile_builder.py

```diff
@@ -1531,21 +1546,24 @@ def save(self, filepath: str = None) -> None:
         :type filepath: String
         :return: None
         """
-        # Create dictionary for all metadata, options, and profile
-        data_dict = {
-            "total_samples": self.total_samples,
-            "sample": self.sample,
-            "encoding": self.encoding,
-            "file_type": self.file_type,
-            "_samples_per_update": self._samples_per_update,
-            "_min_true_samples": self._min_true_samples,
-            "_empty_line_count": self._empty_line_count,
-            "memory_size": self.memory_size,
-            "options": self.options,
-            "_profile": self.profile,
-            "times": self.times,
-        }
-        self._save_helper(filepath, data_dict)
+        if filepath is None or filepath.split(".")[-1] != "json":
```

Comment:
basically if the filepath doesn't end in "json" just default to pkl

Reply:
or if the filepath doesn't exist

Reply:
We might want to use python's pathlib as it auto will get for us properly and handle strange cases like:
```
import pathlib


print(pathlib.Path("hello/foo.bar.tar.gz").suffix)
# output: ".gz"

 print(pathlib.Path("hello/foo.bar.tar.gz/a").suffix)
# output: ""
```
notably the last case which has `.` but not as the ext